### PR TITLE
Adds a new Tech to the B.E.P.I.S., Specialized Engineering, and a new Minor Reward.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Round 2.
image
Adds a new tech to be unlocked within the B.E.P.I.S.'s major techs, called "Specialized Engineering". Within that tech are 2 new items, **Heat Resistant Rods**, and the **Tinker's Gloves**, as well as a new minor tech, the **Survival Pen**.
**Heat Resistant Rods:**
Have you ever wanted to expand Lavaland Base? Build a lava fortress worthy of your magnificence? Well, now you can! Using Heat Resistant Rods, you can make a catwalk to cross lava tiles, as well as build atop lava, without messy methods like using the RCD!
**Tinker's Gloves:**
This just in engineers, insulated gloves have new competition in town! The Tinker's gloves are complicated, overdesigned gloves that, while not very shockproof, allow for faster wall girder construction. No longer will you need to die of old age walling off an area when you're out of RCD ammo! _Warning: This product contains no likeness to clockwork gauntlets used by the extinct cult of Rat'Var, and any such similarity is by no means intentional._

**Survival Pen:**
Have you ever been stuck on lavaland, trapped by your survival pod with just a few chairs, some titanium walls, and walls of ash closing in on you? Well THANK GOD you have your Rockbreaker brand Survival Pen! Allows for basic mining operations, and is portable like a standard pen!
**BUT WAIT!**
A watcher blocks your path. Thankfully, your expensive, diamond encrusted pen isn't just good for being the world's slowest pickaxe, it's ALSO coated in the one thing watchers crave: DIAMONDS. Toss the pen to draw the watcher's attention elsewhere.
## Why It's Good For The Game
 
Starts to fill our B.E.P.I.S. Major Reward Techs, as intended, and fills a niche that doesn't get much play nowadays: Building on Lavaland. This was something I've had in the back of my mind since we did the first tests of the Disaster gamemode a few months ago, so each item here was made with the intention of improving the mining base should the station become un-livable. Lava-Proof Rods are beneficial in that you'll be able to cross and build over lava by the mid-end of the shift. RCDs still work as usual, but this way you won't have to worry about getting specialized engineering equipment as a member of cargo or science. The tinker's gloves are basically a variant of the nitrile gloves, but they grant a speed bonus to adding plating to metal girders, so that some kind of functional alternative to giving every engineer insulated gloves.
Survival Pens are quite honestly very niche, but getting dumb pens with extra functionality is a tradition of trade shows all over the world, so it feels right at home as a minor reward.

Also, does a tiny change to the doe sprites, just to look a little bit less old.
## Changelog
:cl:
add: A new Technology has been implemented as a major reward in the B.E.P.I.S., Specialized Engineering, to appeal to engineering utility and new construction horizons.
tweak: Watchers will now actively consume diamond ore left lying around, alongside the new survival pens.
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
good stuff, not really sure about the pen though.

**Comment by trollbreeder** [comment]
> good stuff, not really sure about the pen though.

The pen is mightier than the sword.

**Comment by wesoda25** [comment]
haha sweet

File: code/modules/research/designs/misc_designs.dm [hunk 1]
```diff
@@ -284,7 +284,26 @@
 	materials = list(/datum/material/iron=200, /datum/material/glass = 100)
 	build_path = /obj/item/card/id
 	category = list("Electronics")
-	departmental_flags = DEPARTMENTAL_FLAG_SERVICE	
+	departmental_flags = DEPARTMENTAL_FLAG_SERVICE
+
+/datum/design/eng_gloves
+	name = "Tinkers Gloves"
+	desc = "Overdesigned engineering gloves that have automated construction subrutines dialed in, allowing for faster construction while worn."
```
Comment (Hatterhat) [inline code/modules/research/designs/misc_designs.dm]: *subroutine

**Comment by HomocidlePancakes** [comment]
Cocaine gloves

**Comment by RaveRadbury** [comment]
This is dank, thanks for this. I'll have to give the Disaster mode another shot in the future.

**Comment by FantasticFwoosh** [comment]
You can already add rods above pits and then use floortiles since goof optimised it, but catwalks are welcome.

**Comment by nightred** [comment]
With this new tech rods, and rcd (except the admin version) should not be allowed to be used on lava.

**Comment by ArcaneMusic** [comment]
> With this new tech rods, and rcd (except the admin version) should not be allowed to be used on lava.

Thing is, there's no guarantee that you will get this tech going into the late-game, and the philosophy behind adding content as a tech in the B.E.P.I.S. is that it's content that it doesn't replace existing lategame content, but instead provides a side-grade to existing methods. As it stands, RCDs can be used to cross lava as early as you can get them, but they're still exceptionally inefficient with materials. Using lava rods would give you a slower, but more "efficient" method of building or crossing lava, at the expense of needing a specialized building material to do so.

**Comment by ArcaneMusic** [comment]
Alright, so I did some retooling for stackcrafting in the process. Now when stackcrafting is called, there's 2 new variables that are checked in the process, **trait_booster** and **trait_modifier**. if the user has the trait_booster trait, their build time is multiplied by trait_modifier. 
I got it so the TRAIT_QUICK_BUILD given by tinker's gloves is applied to building girders, and makes the process a timer of 30 instead of 40 (3 Seconds down from 4), which is notable when you have to build several walls by hand manually.
This does expand the scope of the PR slightly, but all in all it's still a fairly minor change.

**Comment by kriskog** [comment]
You are failing the checks

**Pull request merged**